modules/richards/src/actions/Q2PAction.C

```diff
@@ -100,8 +100,8 @@ Q2PAction::Q2PAction(const InputParameters & params)
     _gas_relperm(getParam<UserObjectName>("gas_relperm")),
     _gas_viscosity(getParam<Real>("gas_viscosity")),
     _diffusivity(getParam<Real>("diffusivity")),
-    _output_nodal_masses_to(getParam<std::vector<OutputName>>("output_nodal_masses_to")),
-    _output_total_masses_to(getParam<std::vector<OutputName>>("output_total_masses_to")),
+    _output_nodal_masses_to(isParamValid("output_nodal_masses_to")? getParam<std::vector<OutputName>>("output_nodal_masses_to"):std::vector<OutputName>{}),
+    _output_total_masses_to(isParamValid("output_total_masses_to")? getParam<std::vector<OutputName>>("output_total_masses_to"):std::vector<OutputName>{}),
```

Comment:
I disagree with this fix pattern. Please supply a default argument (empty vector) instead. This logic is reimplementing the default in a (for the documentation and autocomplete) intransparent way.